partiql-lang/src/main/kotlin/org/partiql/lang/eval/physical/PhysicalPlanCompilerAsyncImpl.kt

```diff
@@ -0,0 +1,1908 @@
+            is PartiqlPhysical.Expr.Lte -> compileLte(expr, metas)
+            is PartiqlPhysical.Expr.Like -> compileLike(expr, metas)
+            is PartiqlPhysical.Expr.InCollection -> compileIn(expr, metas)
+
+            // logical operators
+            is PartiqlPhysical.Expr.Ne -> compileNe(expr, metas)
+            is PartiqlPhysical.Expr.Or -> compileOr(expr, metas)
+
+            // unary
+            is PartiqlPhysical.Expr.Not -> compileNot(expr, metas)
+            is PartiqlPhysical.Expr.Pos -> compilePos(expr, metas)
+            is PartiqlPhysical.Expr.Neg -> compileNeg(expr, metas)
+
+            // other operators
+            is PartiqlPhysical.Expr.Concat -> compileConcat(expr, metas)
+            is PartiqlPhysical.Expr.Call -> compileCall(expr, metas)
+            is PartiqlPhysical.Expr.NullIf -> compileNullIf(expr, metas)
+            is PartiqlPhysical.Expr.Coalesce -> compileCoalesce(expr, metas)
+
+            // "typed" operators (RHS is a data type and not an expression)
+            is PartiqlPhysical.Expr.Cast -> compileCast(expr, metas)
+            is PartiqlPhysical.Expr.IsType -> compileIs(expr, metas)
+            is PartiqlPhysical.Expr.CanCast -> compileCanCast(expr, metas)
+            is PartiqlPhysical.Expr.CanLosslessCast -> compileCanLosslessCast(expr, metas)
+
+            // sequence constructors
+            is PartiqlPhysical.Expr.List -> compileSeq(ExprValueType.LIST, expr.values, metas)
+            is PartiqlPhysical.Expr.Sexp -> compileSeq(ExprValueType.SEXP, expr.values, metas)
+            is PartiqlPhysical.Expr.Bag -> compileSeq(ExprValueType.BAG, expr.values, metas)
+
+            // bag operators
+            is PartiqlPhysical.Expr.BagOp -> compileBagOp(expr, metas)
+            is PartiqlPhysical.Expr.BindingsToValues -> compileBindingsToValues(expr)
+            is PartiqlPhysical.Expr.Pivot -> compilePivot(expr, metas)
+            is PartiqlPhysical.Expr.GraphMatch -> TODO("Physical compilation of GraphMatch expression")
+            is PartiqlPhysical.Expr.Timestamp -> TODO()
+        }
+    }
+
+    private suspend fun compileBindingsToValues(expr: PartiqlPhysical.Expr.BindingsToValues): PhysicalPlanThunkAsync {
+        val mapThunk = compileAstExpr(expr.exp)
+        val bexprThunk: RelationThunkEnvAsync = bexperConverter.convert(expr.query)
+
+        val relationType = when (expr.metas.containsKey(IsOrderedMeta.tag)) {
+            true -> RelationType.LIST
+            false -> RelationType.BAG
+        }
+
+        return thunkFactory.thunkEnvAsync(expr.metas) { env ->
+            // we create a snapshot for currentRegister to use during the evaluation
+            // this is to avoid issue when iterator planner result
+            val currentRegister = env.registers.clone()
+            val elements: Flow<ExprValue> = flow {
+                env.load(currentRegister)
+                val relItr = bexprThunk(env)
```

Comment:
This thunk evaluation is an async call. In the synchronous code, this call was within a sequence -- https://github.com/partiql/partiql-lang-kotlin/blob/plan-eval-async-keep-sync/partiql-lang/src/main/kotlin/org/partiql/lang/eval/physical/PhysicalPlanCompilerImpl.kt#L335-L341. But since the evaluation is an async call, we need to use a coroutine `Flow` (similar to a sequence), which allows for an async call within the block.

More on `Flow`s can be found here: https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-flow/.